src/test/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/PileupReadErrorCorrectorUnitTest.java

```diff
@@ -0,0 +1,74 @@
+package org.broadinstitute.hellbender.tools.walkers.haplotypecaller;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMReadGroupRecord;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.testng.Assert.*;
+
+public class PileupReadErrorCorrectorUnitTest {
+
+    @Test
+    public void TestErrorCorrection() {
+        final String refChunk = "GCATAAACATGGCTCACTGC";
+        final String refChunkHard = "AGCCTTGAACTCCTGGGCTCAAGTGATCCTCCTGCCTCAGTTTCCCATGTAGCTGGGACCACAGGTGGGGGCTCCACCCCTGGCTGATTTTTTTTTTTTTTTTTTTTTGAGATAGGGT";
+
+        final String sample = "sample";
+        final SAMReadGroupRecord rg = new SAMReadGroupRecord("rgID");
+        rg.setSample(sample);
+        rg.setPlatform("illumina");
+        final SAMFileHeader header = ArtificialReadUtils.createArtificialSamHeaderWithReadGroup(rg);
+
+        final int numGoodReads = 500;
+        final int numBadReads = 10;
+        final int readLength = 15;
+        final List<GATKRead> finalizedReadList = new ArrayList<GATKRead>(numGoodReads);
+        int offset = 0;
+        final byte[] quals = new byte[readLength];
+
+        Arrays.fill(quals,(byte)30);
+
+        for (int k=0; k < numGoodReads; k++) {
+            final byte[] bases = Arrays.copyOfRange(refChunk.getBytes(),offset,offset+readLength);
+            final GATKRead read = ArtificialReadUtils.createArtificialRead(header, bases, quals,readLength+"M");
+            read.setName("good_read" + k);
+            read.setReadGroup(rg.getReadGroupId());
```

Comment:
I'm surprised that LocusIteratorByState works without start positions, let alone unsorted.

Reply:
I think all of these start at the same position (1:10000). I think you intended to slide the start position by `offset`. I think adding the start positions to each read should do the trick (and then sort before you pass the list to LocusIteratorByState).

Reply:
done + done